### PR TITLE
feat: support mono repos

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**
scan-configs now runs as a dedicated cloud run job.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/repo-automation-bots/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: packages/owl-bot/bootstrap-scan-configs.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    --concurrency=1 \
+    --memory=4Gi \
+    --ingress=all \
+    --max-instances=1 \
+    --min-instances=0 \
+    --port=8080 \
+    --service-account=owlbot-scan-configs@repo-automation-bots.iam.gserviceaccount.com \
+    --timeout=59m \
+    --update-secrets=/secrets/github.pem=owlbot_github_key:latest \
+    --args="scan-configs,--pem-path,/secrets/github.pem,--app-id,99011,--installation,14695777,--project,repo-automation-bots-metrics,--org,googleapis,--port,8080"
```
Comment (tmatsuo) [inline packages/owl-bot/bootstrap-scan-configs.sh:40]: Note: These app-id and installation id, and the projects are hard coded and need to parameterize when deploying to the staging environment.

File: packages/owl-bot/bootstrap-scan-configs.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    --filter 'owlbot-cli-mono-repo-test')
+
+gcloud scheduler jobs update http invoke-owlbot-cli-mono-repo-test \
+    --project repo-automation-bots \
+    --schedule="5 5 * * *" \
+    --uri="${URL}/scan-configs" \
+    --http-method=GET \
+    --attempt-deadline=30m \
+    --time-zone=America/Los_Angeles \
+    --oidc-service-account-email 856896688174-compute@developer.gserviceaccount.com
```
Comment (tmatsuo) [inline packages/owl-bot/bootstrap-scan-configs.sh:55]: Eventually the service account needs to be parameterized.

File: packages/owl-bot/bootstrap-scan-configs.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+# values will be updated for production:
+#  1. Change the name of the deployment to owlbot-cli-scan-configs
+#  2. Change to --image=gcr.io/repo-automation-bots/owlbot-cli:latest
+
+# I see no need to make updating scan-configs automatically.  It will rarely
+# change, and when that happens, a user can manually re-run this script.
+
+set -ex
+
+gcloud beta run deploy owlbot-cli-mono-repo-test \
```
[the remark this answers is not in the document]
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/bootstrap-scan-configs.sh]: Done.

File: packages/owl-bot/bootstrap-scan-configs.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    --project repo-automation-bots \
+    --image=gcr.io/repo-automation-bots/owlbot-cli-mono-repo-test:v4 \
```
[the remark this answers is not in the document]
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/bootstrap-scan-configs.sh]: Done.  It actually uses the same owlbot-cli we already build.

File: packages/owl-bot/bootstrap-scan-configs.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    --update-secrets=/secrets/github.pem=owlbot_github_key:latest \
+    --args="scan-configs,--pem-path,/secrets/github.pem,--app-id,99011,--installation,14695777,--project,repo-automation-bots-metrics,--org,googleapis,--port,8080"
+
+URL=$(gcloud run services list \
+    --project repo-automation-bots \
+    --platform managed \
+    --format 'value(status.url)' \
+    --filter 'owlbot-cli-mono-repo-test')
+
+gcloud scheduler jobs update http invoke-owlbot-cli-mono-repo-test \
```
[the remark this answers is not in the document]
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/bootstrap-scan-configs.sh]: Done.

File: packages/owl-bot/bootstrap-scan-configs.sh (hunk 1 above)
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/bootstrap-scan-configs.sh:40]: Acknowleged.

File: packages/owl-bot/bootstrap-scan-configs.sh (hunk 2 above)
Comment (SurferJeffAtGoogle) [reply packages/owl-bot/bootstrap-scan-configs.sh:55]: Acknowleged.

**Comment by SurferJeffAtGoogle** [comment]
Ready for another look.

**Review by tmatsuo — APPROVED** [review]

File: packages/owl-bot/bootstrap-scan-configs.sh [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+#  2. Change to --image=gcr.io/repo-automation-bots/owlbot-cli:latest
+
+# I see no need to make updating scan-configs automatically.  It will rarely
+# change, and when that happens, a user can manually re-run this script.
+
+set -ex
+
+gcloud beta run deploy owlbot-scan-configs \
+    --project repo-automation-bots \
+    --image=gcr.io/repo-automation-bots/owlbot-cli \
```
Comment (tmatsuo) [inline packages/owl-bot/bootstrap-scan-configs.sh:29]: Is the image `owlbot-cli` intentional?

Comment (SurferJeffAtGoogle) [reply packages/owl-bot/bootstrap-scan-configs.sh:29]: Yes.

**Pull request merged**